### PR TITLE
Issue 182 body element needs id

### PR DESCRIPTION
**Pull request by Derek52**
I ran into an issue here, https://github.com/Derek52/kweb-core/blob/ee2f1c7cb09225eb3f28651192d264bc1a50c9c1/src/main/kotlin/kweb/html/Document.kt#L19-L40

Here, Document.kt is creating a Body and a Head element. Now that id's are non-nullable they need id's passed to them. But, I assign an id to these elements in HtmlDocumentSupplier like we talked about. I would think we need the same id passed to these elements in both source files, and I'm not seeing a way to do that. I imagine this might have factored into the decision to not include the id's in the first place.

### CODE REVIEW

File: src/main/kotlin/kweb/Element.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ open class Element(
         val creator: ElementCreator<*>?,
         @Volatile open var jsExpression: String,
```
Comment (sanity) [inline src/main/kotlin/kweb/Element.kt:27]: Can we remove this now that id is required?

File: src/main/kotlin/kweb/html/HtmlDocumentSupplier.kt [hunk 2]
```diff
@@ -44,6 +48,10 @@ internal object HtmlDocumentSupplier {
         }
     }
 
+    fun generateId() : String {
```
Comment (sanity) [inline src/main/kotlin/kweb/html/HtmlDocumentSupplier.kt:51]: Should this be private? Not sure where it's being used, but may belong in util/misc.kt.

**Review by sanity — COMMENTED** [review]
Made a few comments but overall looks good. Looks like the build failed for continuous integration but may have been an intermitted issue related to dependencies, I've triggered a rebuild.

File: src/main/kotlin/kweb/html/HtmlDocumentSupplier.kt (hunk 2 above)
Comment (Derek52) [reply src/main/kotlin/kweb/html/HtmlDocumentSupplier.kt:51]: It's currently only being used in this file, so I should have made it private. We may need to move the id generation somewhere else though, so this function might get moved.

File: src/main/kotlin/kweb/Element.kt (hunk 1 above)
Comment (Derek52) [reply src/main/kotlin/kweb/Element.kt:27]: I think so. I'll have to do that on my branch where I'm working on function caching though.

**Pull request merged**